react/features/video-layout/functions.js

```diff
@@ -35,30 +35,49 @@ export function getMaxColumnCount() {
 }
 
 /**
- * Returns the cell count dimensions for tile view. Tile view tries to uphold
- * equal count of tiles for height and width, until maxColumn is reached in
- * which rows will be added but no more columns.
+ * Returns the cell count dimensions for tile view. Tile view tries to fill
+ * the available space optimally
  *
  * @param {Object} state - The redux store state.
- * @param {number} maxColumns - The maximum number of columns that can be
- * displayed.
  * @returns {Object} An object is return with the desired number of columns,
  * rows, and visible rows (the rest should overflow) for the tile view layout.
  */
-export function getTileViewGridDimensions(state: Object, maxColumns: number = getMaxColumnCount()) {
+export function getTileViewGridDimensions(state: Object) {
     // When in tile view mode, we must discount ourselves (the local participant) because our
     // tile is not visible.
     const { iAmRecorder } = state['features/base/config'];
     const numberOfParticipants = state['features/base/participants'].length - (iAmRecorder ? 1 : 0);
+    const { clientHeight, clientWidth } = state['features/base/responsive-ui'];
+    const aspectRatio = state['features/base/responsive-ui'].aspectRatio;
 
-    const columnsToMaintainASquare = Math.ceil(Math.sqrt(numberOfParticipants));
-    const columns = Math.min(columnsToMaintainASquare, maxColumns);
-    const rows = Math.ceil(numberOfParticipants / columns);
-    const visibleRows = Math.min(maxColumns, rows);
+    let bestColumns = 0;
+    let bestRows = 0;
+    let bestTileWidth = 0;
+    let bestFound = false;
+
+    while (!bestFound) {
+        const columns = bestColumns + 1;
+        const straightTileWidth = Math.floor(clientWidth / columns);
+
+        const rows = Math.ceil(numberOfParticipants / columns);
+        const straightTileHeight = Math.floor(clientHeight / rows);
+
+        const constrainedTileWidth = Math.ceil(straightTileHeight * aspectRatio);
```

Comment:
aspectRatio is a Symbol. Have you tested this code?

Reply:
No, I have not been able to do so, as I do not have a dev environment set up for this codebase. It was much quicker to just whip up some code than to get the dev env set up. When I was faced with the bot telling me to sign a legal document it also demotivated me from putting in that work as I wasn't sure you would even consider this PR. Your review gives me indication that it's worth it after all to look into that, so hopefully I'll get it working soon :)

Reply:
Thank you!

Reply:
Resolving this thread as I have changed the aspectRatio referenced. Will still look into dev/testing env.